packages/network-controller/src/NetworkController.ts

```diff
@@ -203,9 +218,9 @@ export class NetworkController extends BaseControllerV2<
     this.lookupNetwork();
   }
 
-  private registerProvider() {
-    this.provider.on('error', this.verifyNetwork.bind(this));
-    this.ethQuery = new EthQuery(this.provider);
+  private registerProvider(provider: ProviderEngine) {
```

Comment:
Nit: What motivated these changes to `registerProvider` and `updateProvider`? This seems functionally equivalent.

Reply:
Ohhh I see. Equivalent JS, but TypeScript demanded it. Makes sense.